src/MongoDB.Driver/IFindFluent.cs

```diff
@@ -53,15 +54,31 @@ public interface IFindFluent<TDocument, TProjection> : IAsyncCursorSource<TProje
         /// </summary>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The count.</returns>
+        [Obsolete("Use CountDocuments instead.")]
         long Count(CancellationToken cancellationToken = default(CancellationToken));
 
         /// <summary>
         /// Counts the number of documents.
         /// </summary>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>A Task whose result is the count.</returns>
+        [Obsolete("Use CountDocumentsAsync instead.")]
         Task<long> CountAsync(CancellationToken cancellationToken = default(CancellationToken));
 
+        /// <summary>
+        /// Counts the number of documents.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The count.</returns>
+        long CountDocuments(CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary>
+        /// Counts the number of documents.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A Task whose result is the count.</returns>
+        Task<long> CountDocumentsAsync(CancellationToken cancellationToken = default(CancellationToken));
+
```

Comment:
Note: NO EstimatedDocumentCount on the fluent Find API.

EstimatedDocumentCount is only supported on the IMongoCollection itself.